src/animation.rs

```diff
@@ -0,0 +1,532 @@
+//! [`Clip`]: struct.Clip.html
+//! [`Mixer`]: struct.Mixer.html
+//! [`Mixer::action`]: struct.Mixer.html#method.action
+//! [`Mixer::update`]: struct.Mixer.html#method.update
+
+use cgmath;
+use froggy;
+use mint;
+use std::sync::mpsc;
+
+use mint::IntraXYZ as IntraXyz;
+use Object;
+
+/// Describes the interpolation behaviour between keyframes.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum Interpolation {
+    /// Immediate change between keyframe values.
+    Discrete,
+
+    /// Linear interpolation between keyframe values.
+    Linear,
+
+    /// Smooth cubic interpolation between keyframe values.
+    Cubic,
+
+    /// Smooth Catmull–Rom spline interpolation between keyframe values.
+    CatmullRom,
+}
+
+/// Describes the looping behaviour of an [`Action`].
+///
+/// [`Action`]: struct.Action.html
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum LoopMode {
+    /// Play the clip in forward order exactly once, i.e. do not loop at all.
+    Once,
+
+    /// Play the clip in forward order, repeating from the start.
+    Repeat {
+        /// The maximum number of repetitions.
+        ///
+        /// When set to `None`, the loop will repeat indefinately.
+        limit: Option<u32>,
+    },
+
+    /// Play the clip alternatively in forward and reverse order.
+    PingPong {
+        /// The maximum number of repetitions.
+        ///
+        /// When set to `None`, the loop will repeat indefinately.
+        limit: Option<u32>,
+    },
+}
+
+/// Describes the target property of an animation.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum Binding {
+    /// Targets the position property of an [`Object`].
+    ///
+    /// The corresponding keyframe values must be [`Vector3`].
+    ///
+    /// [`Object`]: ../object/struct.Object.html
+    /// [`Vector3`]: enum.Values.html#variant.Vector3
+    Position,
+
+    /// Targets the orientation property of an [`Object`].
+    ///
+    /// The corresponding keyframe values must be [`Quaternion`] or [`Euler`].
+    ///
+    /// [`Object`]: ../object/struct.Object.html
+    /// [`Quaternion`]: enum.Values.html#variant.Quaternion
+    /// [`Euler`]: enum.Values.html#variant.Euler
+    Orientation,
+
+    /// Targets the scale property of an [`Object`].
+    ///
+    /// The corresponding keyframe values must be [`Scalar`].
+    ///
+    /// [`Object`]: ../object/struct.Object.html
+    /// [`Scalar`]: enum.Values.html#variant.Scalar
+    Scale,
+}
+
+/// An index into the frames of a track.
+enum FrameRef {
+    /// The time is before the start of the frames.
+    Unstarted,
+
+    /// The time corresponds to the given frame index.
+    InProgress(usize),
+
+    /// The time is after the end of the last frame.
+    Ended,
+}
+
+/// The keyframe values of a [`Track`].
+///
+/// [`Track`]: struct.Track.html
+#[derive(Clone, Debug)]
+pub enum Values {
+    /// Euler angle keyframes in radians.
+    Euler(Vec<mint::EulerAngles<f32, IntraXyz>>),
```

Comment:
Corresponds to `cgmath::EulerAngles`. Is this the most common representation?

Reply:
Can't answer that right now. Let's revisit it later.